### PR TITLE
Refine useAnimations feature

### LINKED ISSUE #206: [Refine feature] useAnimations impl on multiplayer
Hi there👋

Sadly I played the Minecraft Bedrock Edition (is MC for Windows 1.19.51), its behavior is Java like.

* Hotbar item is not bobbing when attacking the entity.
* Also when breaking the block, Consuming foods.

However, I really like this animation, so I have made it available on multiplayer.
I’m going to test it on my vanilla server. I think it also works on Realms if it’s okay.
I’ll create a new PR when ready, about a few days.

Thank you for awesome mod!

----

**Behavior**

The bobbing animation now execute on Render thread only.

The hotbar item will animate when:
* Attacking the entity even if weapons / tools gets no damage.
* Breaking the block.
* Using tools.
* Accessing the block / fluid.
* Decreasing in stack.
* The item stack has been changed, such as catching a fish into a bucket.
* Blocking the attacks with the Shield.
* Mending by xp orb.

**Known issue**

* Non-mining tools such as “Flint and Steel” are always animated when destroying a block using it.
* May cause a double-bobbing. e.g.) Placing any blocks, Extinguishing a campfire with a shovel.
* Extinguishing a lit candle on a Candle Cake block does not animate.

----

**For Developers**

The final stage Git branch is available on my repo. If you interested in, you can build and test it. (And I’d be happy you find and report a bug in this thread.)

~[lonefelidae16:feat/client-use-anim](https://github.com/lonefelidae16/BedrockIfy/tree/feat/client-use-anim)~
edit: Successfully merged into upstream

### PR DESCRIPTION
**Pull request by lonefelidae16**
close #206 

Bobbing animation with S2C network interaction.
It also works when played on a vanilla server.

**Known issues**

* ~Non-mining tools such as “Flint and Steel” are always animated when destroying a block using it.~
  Solved.
* ~Using Bone Meal will cause double-bobbing due to `ClientPlayerInteractionManager#interactItem` returns `ActionResult.SUCCESS` and decreasing in stack.~
  edit: Solved in commit: c2395f621c8bf7c71aad0ddf25f59152127f7bfe and da10b9d8459c6ce17f34f178955a124481b0489a.
* Firing Arrows with a Bow / Crossbow wont do bobbing animation until they get damage because there is no status to handle.
  - It is possible to define a brand new status in `LivingEntity` and handle it in `ClientPlayerEntity`, but this solution doesn’t work on a vanilla server. Also, there may be a conflict in the magic number of the statuses in a future update.
* Extinguishing a lit candle still does not animate.

### CODE REVIEW

File: src/main/java/me/juancarloscp52/bedrockify/client/features/useAnimations/AnimationsHelper.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            return;
+        }
+        if (target.isEmpty()) {
+            return;
+        }
+
+        final long now = System.currentTimeMillis();
+        if (ItemStack.areEqual(target, lastBobbingStack) && lastBobbingTime + COOLDOWN_TIME_MILLIS > now) {
+            return;
+        }
```
Comment (lonefelidae16) [inline src/main/java/me/juancarloscp52/bedrockify/client/features/useAnimations/AnimationsHelper.java]: Apparently, this `if` statement is always false. I’ll revert 88f4949f29d7c633dcbcf41552db5863d72eac7c.

**Comment by lonefelidae16** [comment]
I will end my commit with this comment for now.

Works fine for me (and I enjoying on my server), but there are still issues:

* Firing Arrows with a Bow / Crossbow do not animate until they get damage.
* May cause a double-bobbing using tools. e.g.) Destroying any blocks, Extinguishing a campfire.
  - If this had prevented, mending tools by xp orb doesn’t animate.
* Extinguishing a lit candle using tools still doesn’t animate.

Sorry for the many commits. If I should squash these, it doesn’t matter to combine them into a reasonable amount of granularity. But please note, it needs to FORCE push with `-f` option.

Please let me know if you have any further suggestions.

**Comment by juancarloscp52** [comment]
Hey! Sorry for the late response. I'll be testing this implementation and its differences with the previous one this week

**Comment by lonefelidae16** [comment]
Thank you! Whenever you’re ready.

**Comment by juancarloscp52** [comment]
Looks good! I think that the remaining issues are not as noticeable, and is a comprehensible compromise for having item animations client-side.

Thank you.

**Pull request merged**